### PR TITLE
[CI] Improve management of locked dependencies

### PR DESCRIPTION
**Pull request by ghisvail**
- Add target to check the consistency of the locked dependencies with the project
- Ensure install is run with a consistent lock file
- Add target to refresh the lock file

⚠️  Requires Poetry version 1.2

### CODE REVIEW

**Comment by ghisvail** [comment]
Yes, Jenkins is still stuck with Poetry v1.1 🤦

**Comment by ghisvail** [comment]
Ideally, we'd want Poetry to be available separately from the project's environment like it's done with GitHub CI. The recommended way nowadays is to use `pipx` to install Python tools into their separate environment and avoid interferences. We should probably adopt the same approach for our CI.

**Comment by ghisvail** [comment]
To be rebased onto #765

**Comment by ghisvail** [comment]
> `poetry==1.2.1` is available in the CI agents.

I get `make: poetry: Command not found` after removal of poetry from Conda manifests.

**Review by NicolasGensollen — APPROVED** [review]
Great ! Thanks ! 👍

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ pipeline {
   agent none
   stages {
     stage('Checkout') {
-      failFast true
+      //failFast true
```
Comment (NicolasGensollen) [inline .jenkins/Jenkinsfile:9]: Maybe we can undo this before merging

**Pull request merged**